datasets/bnl_newspapers/bnl_newspapers.py

```diff
@@ -0,0 +1,188 @@
+        **kwargs: keyword arguments forwarded to super.
+        """
+        super(BNLNewspapersConfig, self).__init__(version=datasets.Version("1.17.0"), **kwargs)
+        self.data_url = data_url
+        self.citation = citation
+        self.url = url
+
+
+class BNLNewspapers(datasets.GeneratorBasedBuilder):
+    """Historic newspapers from the BNL"""
+
+    BUILDER_CONFIGS = [
+        BNLNewspapersConfig(
+            name="processed",
+            description="Processed BNL newspapers",
+            data_url=_URLs["processed"],
+            citation=_CITATION,
+            url=_HOMEPAGE,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "processed"
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "id": datasets.Value("string"),
+                "source": datasets.Value("string"),
+                "url": datasets.Value("string"),
+                "title": datasets.Value("string"),
+                "ispartof": datasets.Value("string"),
+                "text": datasets.Value("string"),
+                "pub_date": datasets.Value("timestamp[s]"),
+                "publisher": datasets.Value("string"),
+                "language": datasets.Value("string"),
+                "article_type": datasets.ClassLabel(
+                    names=[
+                        "ADVERTISEMENT_SECTION",
+                        "BIBLIOGRAPHY",
+                        "CHAPTER",
+                        "INDEX",
+                        "CONTRIBUTION",
+                        "TABLE_OF_CONTENTS",
+                        "WEATHER",
+                        "SHIPPING",
+                        "SECTION",
+                        "ARTICLE",
+                        "TITLE_SECTION",
+                        "DEATH_NOTICE",
+                        "SUPPLEMENT",
+                        "TABLE",
+                        "ADVERTISEMENT",
+                        "CHART_DIAGRAM",
+                        "ILLUSTRATION",
+                        "ISSUE",
+                    ]
+                ),
+                "extent": datasets.Value("int32"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
```

Comment:
```suggestion
            task_templates=[LanguageModeling(text_column="text")]
        )
```

This change requires regenerating the dataset_infos.json file.